stlab/concurrency/future.hpp

```diff
@@ -1810,13 +1812,13 @@ auto shared_base<T, enable_if_copyable<T>>::reduce(future<future<R>>&& r) -> fut
 
 template <typename T>
 auto shared_base<T, enable_if_not_copyable<T>>::reduce(future<future<void>>&& r) -> future<void> {
-    return std::move(r).then([](auto&&){});
+    return std::move(r).then([](auto){});
 }
 
 template <typename T>
 template <typename R>
 auto shared_base<T, enable_if_not_copyable<T>>::reduce(future<future<R>>&& r) -> future<R> {
-    return std::move(r).then([](auto&& f) { return *std::forward<future<R>>(f).get_try(); });
+    return std::move(r).then([](auto&& f) { return *std::move(f).get_try(); });
```

Comment:
@FelixPetriconi - can you speak a little bit to why you made this change? I would have thought the correct line here would be `std::forward<decltype(f)>(f)`... why is a `move` correct?

Reply:
My mistake. I was looking at the signature from the reduce function and there it is an rvalue reference, that should be moved. But here, as you correctly have noticed, we are in a forwarding reference and your code would be the general solution to this. I will fix this now.